FirebaseRemoteConfig/Tests/Unit/RCNInstanceIDTest.m

```diff
@@ -0,0 +1,400 @@
+                                           firebaseAppName:currentAppName
+                                               googleAppID:currentOptions.googleAppID];
+    dispatch_queue_t queue = dispatch_queue_create(
+        [[NSString stringWithFormat:@"testqueue: %d", i] cStringUsingEncoding:NSUTF8StringEncoding],
+        DISPATCH_QUEUE_SERIAL);
+    _configFetch[i] = OCMPartialMock([[RCNConfigFetch alloc] initWithContent:configContent
+                                                                   DBManager:_DBManager
+                                                                    settings:settings
+                                                                   analytics:nil
+                                                                  experiment:nil
+                                                                       queue:queue
+                                                                   namespace:fullyQualifiedNamespace
+                                                                     options:currentOptions]);
+  }
+}
+
+// Mock instance ID methods.
+- (void)mockInstanceIDMethodForTokenAndIdentity:(nullable NSString *)token
+                                     tokenError:(nullable NSError *)tokenError
+                                       identity:(nullable NSString *)identity
+                                  identityError:(nullable NSError *)identityError {
+  // Mock the instanceID retreival method.
+  id instanceIDMock = OCMPartialMock([FIRInstanceID instanceIDForTests]);
+  OCMStub([instanceIDMock
+      tokenWithAuthorizedEntity:[OCMArg any]
+                          scope:[OCMArg any]
+                        options:[OCMArg any]
+                        handler:([OCMArg
+                                    invokeBlockWithArgs:(token ? token : [NSNull null]),
+                                                        (tokenError ? tokenError : [NSNull null]),
+                                                        nil])]);
+  OCMStub([instanceIDMock
+      getIDWithHandler:([OCMArg invokeBlockWithArgs:(identity ? identity : [NSNull null]),
+                                                    (identityError ? identityError : [NSNull null]),
+                                                    nil])]);
+}
+
+- (void)tearDown {
+  [_DBManager removeDatabaseOnDatabaseQueueAtPath:_DBPath];
+  [[NSUserDefaults standardUserDefaults] removePersistentDomainForName:_userDefaultsSuiteName];
+  [super tearDown];
+}
+
+// Instance ID token is nil. Error is not nil. Verify fetch fails.
+- (void)testNilInstanceIDTokenAndError {
+  NSMutableArray<XCTestExpectation *> *expectations =
+      [[NSMutableArray alloc] initWithCapacity:RCNTestRCNumTotalInstances];
+
+  // Test for each RC FIRApp, namespace instance.
+  for (int i = 0; i < RCNTestRCNumTotalInstances; i++) {
+    // Set the token as nil.
+    [self mockInstanceIDMethodForTokenAndIdentity:nil
```

Comment:
looks like it should be out of the loop.

Reply:
Done.